src/fs_base_fsync.hpp

```diff
@@ -33,6 +33,6 @@ namespace fs
   int
   fdatasync(const int fd)
   {
-    return ::fdatasync(fd);
+    return fs::fdatasync(fd);
```

Comment:
In two locations I had to add explicit namespaces. I have no idea why. This was the error:

```
In file included from src/fsync.cpp:28:
src/fs_base_fsync.hpp:36:12: error: no member named 'fdatasync' in the global namespace; did you mean simply 'fdatasync'?
    return ::fdatasync(fd);
           ^~~~~~~~~~~
           fdatasync
src/fs_base_fsync.hpp:34:3: note: 'fdatasync' declared here
  fdatasync(const int fd)
  ^
1 error generated.
```